### DISCUSSION TITLE
Migrate from v3 to v4

### DISCUSSION THREAD

**Discussion by Sandy2324-bit** [wixtoolset/issues#8835] [Questions]
Hello,

We have a WixBA written in v3 and we want to migrate it into v4 but from the WixToolSet page the information which we got is we have to migrate from .Net framework to .Net core 6. 

Do I need to migrate it to .Net Core 6 ? If YES, what are the namespaces and the libraries are suppose to add how can it be compatible with the other packages which is already present in project.

**Comment by chrpai** [comment]
I'll be honest, I've never had a need to write a custom BA.  But I have ported a lot o code from .NET Framework to .NET.  If your willing to share the BA source with me I'd be willing to look into what's involved as a favor to you.

**Reply by Sandy2324-bit** [reply]
I want to call those methods present in WixBA file but file not getting executed
 
Please find the WixBA file changes,
Namespace: 

using System;
using System.Collections.Generic;
using System.Diagnostics;
using System.IO;
using System.Net;
using System.Text;
using System.Windows.Input;
using Threading = System.Windows.Threading;
using WinForms = System.Windows.Forms;
using Microsoft.Tools.WindowsInstallerXml.Bootstrapper;
using System.Windows;
using System.Threading;
using System.Windows.Markup;
using System.Globalization;

Methods:

public static void PlanLayout()
{
    if (String.IsNullOrEmpty(WixBA.Model.Command.LayoutDirectory))
    {
        WixBA.Model.LayoutDirectory = Directory.GetCurrentDirectory();

        if (WixBA.Model.Command.Display == Display.Full)
        {
            WixBA.Dispatcher.Invoke((Action)delegate()
            {
                WinForms.FolderBrowserDialog browserDialog = new WinForms.FolderBrowserDialog();
                browserDialog.RootFolder = Environment.SpecialFolder.MyComputer;

                browserDialog.SelectedPath = WixBA.Model.LayoutDirectory;
                WinForms.DialogResult result = browserDialog.ShowDialog();

                if (WinForms.DialogResult.OK == result)
                {
                    WixBA.Model.LayoutDirectory = browserDialog.SelectedPath;
                    WixBA.Plan(WixBA.Model.Command.Action);
                }
                else
                {
                    WixBA.View.Close();
                }
            }
            );
        }
    }
    else
    {
        WixBA.Model.LayoutDirectory = WixBA.Model.Command.LayoutDirectory;
        WixBA.Plan(WixBA.Model.Command.Action);
    }
}

private void RenameProductName()
{
    try
    {
        this.Engine.Log(LogLevel.Verbose, $"enter RenameProductName");

        string installationTypeValue = this.Engine.StringVariables[WixInstallationTypeVariable];
        this.Engine.Log(LogLevel.Verbose, $"RenameProductName, productName is {this.Engine.StringVariables[WixBundleNameVariable]}, InstallationType is {installationTypeValue}");
        InstallationType installationType = InstallationType.None;
        if (Enum.TryParse<InstallationType>(installationTypeValue, true, out installationType))
        {
            switch (installationType)
            {
                case InstallationType.AgentService:
                    this.Engine.StringVariables[WixBundleNameVariable] = WixConstants.WinServiceAppName;
                    break;
				case InstallationType.Plugin:
					this.Engine.StringVariables[WixBundleNameVariable] = WixConstants.SAUpdatePluginName;
				 	break;
					case InstallationType.OSRI:
					this.Engine.StringVariables[WixBundleNameVariable] = WixConstants.OSRIName;
					break;	
                default:
                    break;
            }
        }
        else
        {
            this.Engine.Log(LogLevel.Verbose, "RenameProductName, convert InstallationType failed!");
        }
    }

In Bundle.wxs file:

	<Bundle Name="$(var.ProductName)" Version="$(VersionNumber)" Manufacturer="$(var.Company)" UpgradeCode="SOME_CODE" IconSourceFile="Imageforthat.ico" DisableModify="yes">
		<Variable Name="SuppressDowngradeFailure" Type="string" Value="yes"/>
		<BootstrapperApplication>
			<bal:WixStandardBootstrapperApplication Theme="hyperlinkLicense" LicenseUrl="" LogoFile="LOGO.ico" SuppressOptionsUI="true" SuppressRepair="yes"/>
			<Payload Name="BootstrapperCore.config" SourceFile="$(var.BADir)\WixBA.BootstrapperCore.config" />
			<Payload SourceFile="$(var.BADir)\WixBA.dll" />
